specification/monitor/data-plane/preview/2018-09-01-preview.json

```diff
@@ -0,0 +1,1927 @@
+      "post": {
+        "summary": "Ingestion API used by Azure custom metrics\r\n            Responsible of below operation\r\n            1. Authorize the request\r\n            2. Deserialize payload request body\r\n            3. Perform validation of the payload\r\n            4. Commits the payload for metrics ingestion",
+        "operationId": "postSubscriptionsBySubscriptionIdResourcegroupsByResourceGroupNameProvidersByResourceProviderByResourceTypeNameByResourceNameMetrics",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "header",
+            "name": "Content-Type",
+            "description": "Supports application/json and application/x-ndjson",
+            "type": "string"
+          },
+          {
+            "in": "header",
+            "name": "Content-Length",
+            "description": "Content length of the payload",
+            "type": "integer",
+            "format": "int32"
+          },
+          {
+            "in": "header",
+            "name": "Authorization",
+            "description": "Authorization token issue for issued for audience https:\\\\monitoring.azure.com",
+            "type": "string"
+          },
+          {
+            "in": "path",
+            "name": "subscriptionId",
+            "description": "The azure subscription id",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "path",
+            "name": "resourceGroupName",
+            "description": "The ARM resource group name",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "path",
+            "name": "resourceProvider",
+            "description": "The ARM resource provider name",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "path",
+            "name": "resourceTypeName",
+            "description": "The ARM resource type name",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "path",
+            "name": "resourceName",
+            "description": "The ARM resource name",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "The Azure metrics document json payload",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Microsoft.Azure.Monitoring.Ingestion.Gateway.Models.AzureMetricsDocument"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The azure metrics publish succeeded",
+            "schema": {
+              "$ref": "#/definitions/Microsoft.AspNetCore.Mvc.ObjectResult"
+            }
+          },
+          "400": {
```

Comment:
Shouldn't need to model these failure cases, use ```default``` instead.  See [here](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/datalake-store/data-plane/Microsoft.DataLakeStore/stable/2016-11-01/filesystem.json#L95) for an example.
Also is it expected that the 200 and 4xx cases return the same model?

Reply:
Ideally this information would be returned in the response body as an [OData v4 object](http://docs.oasis-open.org/odata/odata-json-format/v4.0/os/odata-json-format-v4.0-os.html#_Toc372793091).  Something like this.
```json
{
  "Error": {
    "Code": "InvalidRequest",
    "Message": "Some descriptive error message."
  }
}
```

Reply:
We indeed already do that. So is the suggestion that having above behavior good enough?

Reply:
I would suggest then that you specify a ```default``` response that references the appropriate model definition.  Looking at the definition for ```Microsoft.AspNetCore.Mvc.ObjectResult``` it's not an OData v4 error object (or a superset of it).

Reply:
More importantly, the code generators will treat any status code modeled here as a success.  So if these 4xx codes aren't to be treated as success then you will want to remove them.

Reply:
Makes sense. Fixing it. :)